tx-pool/src/service.rs

```diff
@@ -400,11 +402,48 @@ impl TxPoolController {
             .map_err(handle_recv_error)
             .map_err(Into::into)
     }
+
+    /// Saves tx pool into disk.
+    pub fn save_pool(&self) -> Result<(), AnyError> {
+        info!("Please be patient, tx-pool are saving data into disk ...");
+        let mut sender = self.sender.clone();
+        let (responder, response) = oneshot::channel();
+        let request = Request::call((), responder);
+        sender.try_send(Message::SavePool(request)).map_err(|e| {
+            let (_m, e) = handle_try_send_error(e);
+            e
+        })?;
+        self.handle.block_on(response).map_err(handle_recv_error)?
+    }
+
+    fn load_persisted_data(&self, mut txs: Vec<TransactionView>) -> Result<(), AnyError> {
+        info!("Loading persisted tx-pool data, total txs: {}", txs.len());
+        // a trick to commit transactions with the correct order
+        loop {
+            let mut failed_txs = Vec::new();
+            for tx in txs.iter() {
+                if self.submit_local_tx(tx.clone())?.is_err() {
+                    failed_txs.push(tx.clone());
+                }
+            }
+            // return when all success or all failed
+            if failed_txs.is_empty() || txs.len() == failed_txs.len() {
+                info!(
+                    "Persisted tx-pool data is loaded and {} txs are failed to load",
+                    failed_txs.len()
```

Comment:
* check the transaction's status, if the transaction is already known, then it is considered successful.
* Please do not print this log if all is well, as "failed" indicates there is an error

Reply:
> check the transaction's status, if the transaction is already known, then it is considered successful.
I didn't understand the comment, could you be more specific?

> Please do not print this log if all is well, as "failed" indicates there is an error
resolved, please review again.